action/action-util/src/main/groovy/com/powsybl/action/util/GeneratorModificationTaskExtension.groovy

```diff
@@ -0,0 +1,124 @@
+/**
+ * Copyright (c) 2019, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.action.util
+
+import com.google.auto.service.AutoService
+import com.powsybl.action.dsl.spi.DslTaskExtension
+import com.powsybl.commons.PowsyblException
+import com.powsybl.contingency.tasks.ModificationTask
+
+/**
+ * @author Olivier Perrin <olivier.perrin at rte-france.com>
+ */
+@AutoService(DslTaskExtension.class)
+class GeneratorModificationTaskExtension implements DslTaskExtension {
+    @Override
+    void addToSpec(MetaClass tasksSpecMetaClass, List<ModificationTask> tasks, Binding binding) {
+        tasksSpecMetaClass.generatorModification = { String id, Closure<Void> closure ->
+            def cloned = closure.clone()
+            GeneratorModificationSpec spec = new GeneratorModificationSpec()
+            cloned.delegate = spec
+            cloned()
+            if (spec.hasTargetP() && spec.haspDelta()) {
+                throw new PowsyblException("targetP/pDelta actions are both found in generatorModification on '" + id + "'");
+            }
+            tasks.add(new GeneratorModificationTask(id, spec.computeModifs()));
+        }
+    }
+
+    /**
+     * Describes the instructions usable in a "generatorModification" task
+     */
+    static class GeneratorModificationSpec {
+        private Double minP;
+        private Double maxP;
+        private Double targetP;
+        private Double pDelta;
+        private Double targetV;
+        private Double targetQ;
+        private Boolean voltageRegulatorOn;
+        private Boolean connected;
+
+        GeneratorModificationTask.Modifs computeModifs() {
+            GeneratorModificationTask.Modifs modifs = new GeneratorModificationTask.Modifs();
+            modifs.setMinP(minP);
+            modifs.setMaxP(maxP);
+            modifs.setTargetP(targetP);
+            modifs.setpDelta(pDelta);
+            modifs.setTargetV(targetV);
+            modifs.setTargetQ(targetQ);
+            modifs.setVoltageRegulatorOn(voltageRegulatorOn);
+            modifs.setConnected(connected);
+            return modifs;
+        }
+
+        void minP(Double minP) {
```

Comment:
Not needed setter/getter methods on groovy fields, so just remove `private`. It looks more like Groovy-Style.

Reply:
Without these setters, the DSL spec doesn't work.